### PR TITLE
Added `check-smtp-ehlo` for checking whether SMTP is functional

### PR DESCRIPTION
**Pull request by hanynowsky**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [ ] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass 

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,14 +4,15 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+- Add `check-smtp-ehlo` for checking whether SMTP is functional
 
 ## [0.1.0] - 2016-01-29
 ### Added
 - Add support to `check-mailq` for inspecting each Postfix queue individually
 - Add a `check-mail-delay` script to support alerting by age of queue items
 
 ### Fixed
-- metrics-mailq.rb: don't output stderr when the mail queue is empty
+- metrics-mailq.rb: do not output stderr when the mail queue is empty
```
Comment (GhostLyrics) [inline CHANGELOG.md:15]: I think you should remove this, since this PR does not touch said file.

Comment (majormoses) [reply CHANGELOG.md:15]: agreed

**Comment by majormoses** [comment]
@hanynowsky any chance on giving this another go?

**Comment by hanynowsky** [comment]
Sure. Please gimme a week. I am on a hot project.

File: bin/check-smtp-ehlo.rb [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  rescue Net::ReadTimeout => ne
+    critical "Timeout error: #{ne.message}"
+  rescue => e
+    critical "Cannot login to SMTP -  TRACE: #{e.message}"
+  end
+
+  # Main Function
+  def run
+    result = check_ehlo.to_i
+    ok "/opt/sensu/embedded/bin/ruby /etc/sensu/plugins/check-smtp-ehlo.rb \
```
Comment (majormoses) [inline bin/check-smtp-ehlo.rb:129]: remove these hardcoded paths.

File: bin/check-smtp-ehlo.rb [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    --warn-only --domain google.com --hostname localhost --port 25 \
```
Comment (majormoses) [inline bin/check-smtp-ehlo.rb:130]: remove hardcoded domain, hostname, port.

File: bin/check-smtp-ehlo.rb [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+  option(
+    :method,
+    description: 'EHLO or HELO',
+    short: '-m METHOD',
+    long: '--method METHOD',
+    default: 'EHLO'
+  )
+
+  option(
```
Comment (majormoses) [inline bin/check-smtp-ehlo.rb:76]: we don't need this I believe as it should come from the base class plugin.

File: bin/check-smtp-ehlo.rb [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    critical "Cannot login to SMTP -  TRACE: #{e.message}"
+  end
+
+  # Main Function
+  def run
+    result = check_ehlo.to_i
+    ok "/opt/sensu/embedded/bin/ruby /etc/sensu/plugins/check-smtp-ehlo.rb \
+    --warn-only --domain google.com --hostname localhost --port 25 \
+    --method EHLO|HELO \nCURRENT VALUES: #{config.inspect} \n option: \
+    --smtpbind has priority" if config[:help]
```
Comment (majormoses) [inline bin/check-smtp-ehlo.rb:132]: huh?

**Review by majormoses — CHANGES_REQUESTED** [review]

**Comment by majormoses** [comment]
closing due to inactivity, please reply back if you would like to see this through so I can re-open.

**Pull request closed without merging**